modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/StreamLoader.scala

```diff
@@ -48,7 +49,13 @@ object StreamLoader {
 
       val metrics = resources.metrics.report
 
-      load.merge(metrics).compile.drain
+      logger.info(s"BQ Stream Loader ${generated.BuildInfo.version} has started. Listening ${e.config.input.subscription}") *>
+        load.merge(metrics).compile.drain.attempt.flatMap {
+          case Right(_) =>
+            logger.info("Application shutting down") >> IO.pure(ExitCode.Success)
+          case Left(e) =>
+            logger.error(s"Application shutting down with error: $e") *> IO.raiseError(e) >> IO.pure(ExitCode.Error)
```

Comment:
In here it might be better if we don't include exception in the log message at all because we can't control what exception will contain exactly. I am a bit hesitant to include anything in the logs outside of our control after recent log4j CVE. What do you think @istreeter ?

After adding Sentry integration, we will send the exception to Sentry anyway. I guess we don't lose something important even if we don't include exception in the log.

Reply:
I hear what you're saying, but I think we must not be scared to print out exceptions.  And it hurts us if we don't print exceptions.

I really hope we can trust our libraries to not log user-submitted data, and this includes error messages.  But we do just have to trust the libraries on that (and maybe take action later, if we observe them to be breaking those rules).